src/CPU/readTimeOverhead.cpp

```diff
@@ -0,0 +1,51 @@
+// Experiment to measure overhead of reading Read Time-Stamp Counter (RTDSC)
+
+#include <iostream>
+#include <chrono>
+#include <thread>
+#include <unistd.h>
+#include <sched.h>
+#include "timer.hpp"
+#include "stats.hpp"
+#include "proc.hpp"
+using namespace std;
+
+const int N_RUNS = 10;
+const int N_ITERATIONS = 1000;
+
+// Returns number of cycles consumed in reading RTDSC
+double readTimeOverheadInCycles() {
+    Timer t;
+    double n_cycles = 0.0;
+    for (int i = 0; i < N_ITERATIONS; i++) {
+        t.begin();
+        t.end();
+        n_cycles += t.diff();
+    }
+    return n_cycles / N_ITERATIONS;
+}
+
+// Returns time consumed in reading RTDSC 
+double readTimeOverheadInSeconds() {
```

Comment:
maybe rename this to readTimeOverheadInMicrosec

Reply:
Done